### PR TITLE
[EG] Update tests

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py [hunk 1]
```diff
@@ -269,9 +269,11 @@ def test_send_cloud_event_dict(self, eventgrid_cloud_event_topic_endpoint):
         }
         client.send(cloud_event1)
 
+    @pytest.mark.live_test_only
     @EventGridPreparer()
-    @recorded_by_proxy
-    def test_send_signature_credential(self, eventgrid_topic_key, eventgrid_topic_endpoint):
+    def test_send_signature_credential(self, **kwargs):
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py:274]: for here could it be `self(*, eventgrid_topic_key, eventgrid_topic_endpoint)`?

File: sdk/eventgrid/azure-eventgrid/tests/eventgrid_preparer.py [hunk 2]
```diff
@@ -1,10 +1,8 @@
 import functools
-from devtools_testutils import PowerShellPreparer
+from devtools_testutils import EnvironmentVariableLoader
 
 from azure.mgmt.eventgrid.models import Topic, InputSchema, JsonInputSchemaMapping, JsonField, JsonFieldWithDefault
 
-from devtools_testutils.resource_testcase import RESOURCE_GROUP_PARAM
-
 EVENTGRID_TOPIC_PARAM = 'eventgrid_topic'
 EVENTGRID_TOPIC_LOCATION = 'westus'
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/tests/eventgrid_preparer.py:7]: location is still westus so curious why you're changing them to east us?

File: sdk/eventgrid/azure-eventgrid/tests/test_exceptions.py [hunk 3]
```diff
@@ -42,6 +27,7 @@ def create_eg_publisher_client(self, endpoint):
         client = self.create_client_from_credential(EventGridPublisherClient, credential=credential, endpoint=endpoint)
         return client
 
+    @pytest.mark.live_test_only
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/tests/test_exceptions.py:30]: can you remove recorded_by_proxy below?

File: sdk/eventgrid/azure-eventgrid/tests/test_exceptions_async.py [hunk 4]
```diff
@@ -40,6 +29,7 @@ def create_eg_publisher_client(self, endpoint):
         client = self.create_client_from_credential(EventGridPublisherClient, credential=credential, endpoint=endpoint)
         return client
 
+    @pytest.mark.live_test_only
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/tests/test_exceptions_async.py:32]: can you remove recorded_by_proxy?

File: sdk/eventgrid/azure-eventgrid/tests/eventgrid_preparer.py (hunk 2 above)
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/tests/eventgrid_preparer.py:7]: the resources I ran tests with were eastus

**Review by swathipil — APPROVED** [review]

**Pull request merged**